src/main/java/org/apache/maven/plugins/dependency/exclusion/AnalyzeExclusionsMojo.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.maven.plugins.dependency.exclusion;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.function.Consumer;
+
+import org.apache.maven.artifact.Artifact;
+import org.apache.maven.execution.MavenSession;
+import org.apache.maven.model.Dependency;
+import org.apache.maven.plugin.AbstractMojo;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugin.MojoFailureException;
+import org.apache.maven.plugins.annotations.Component;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.plugins.annotations.ResolutionScope;
+import org.apache.maven.project.DefaultProjectBuildingRequest;
+import org.apache.maven.project.MavenProject;
+import org.apache.maven.project.ProjectBuilder;
+import org.apache.maven.project.ProjectBuildingException;
+import org.apache.maven.project.ProjectBuildingRequest;
+import org.apache.maven.project.ProjectBuildingResult;
+
+import static java.lang.String.format;
+import static java.util.stream.Collectors.toList;
+import static java.util.stream.Collectors.toSet;
+import static org.apache.commons.lang3.StringUtils.stripToEmpty;
+import static org.apache.maven.plugins.dependency.exclusion.Coordinates.coordinates;
+
+/**
+ * Analyzes the exclusions defined on dependencies in this project and reports if any of them are invalid.
+ * <p>
+ * Relevant use case is when an artifact in a later version has removed usage of a dependency, making the exclusion no
+ * longer valid.
+ * </p>
+ *
+ * @since 3.6.2
+ */
+@Mojo(name = "analyze-exclusions", requiresDependencyResolution = ResolutionScope.TEST, threadSafe = true)
+public class AnalyzeExclusionsMojo extends AbstractMojo {
+
+    @Component
+    private MavenProject project;
+
+    @Component
+    private ProjectBuilder projectBuilder;
+
+    @Component
+    private MavenSession session;
+
+    /**
+     * Whether to fail the build if invalid exclusions is found.
+     */
+    @Parameter(property = "failOnWarning", defaultValue = "false")
+    private boolean failOnWarning;
```

Comment:
Last comment - I would like to change name of this parameter to something else. 
The same is used in `analyze` gola, so I will can not define different values for both

Reply:
I mentioned in the first post that I reused the property. I am not sure if there are any maven naming standards for these kind of properties. 

"mdep.exclusion.failOnWarning"? I have no strong feelings here, except maybe it shouldn't be too long?

Reply:
my proposition:

```java
@Parameter(property = "mdep.exclusion.fail", defaultValue = "false")
private boolean exclusionFail;
```

Reply:
Updated

Reply:
I see that you update `skip` parameter 😄

Reply:
Whoops, I am too tired to even operate light machinery. Corrected it now, thanks.  :)